kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -699,13 +681,19 @@ impl<'tcx> GotocCtx<'tcx> {
                             // DISCRIMINANT - val:0 ty:i8
                             // DISCRIMINANT - val:1 ty:i8
                             let discr = Expr::int_constant(discr.val, self.codegen_ty(discr_t));
-                            unwrap_or_return_codegen_unimplemented_stmt!(
+                            let place_goto_expr = unwrap_or_return_codegen_unimplemented_stmt!(
                                 self,
                                 self.codegen_place(place)
                             )
-                            .goto_expr
-                            .member("case", &self.symbol_table)
-                            .assign(discr, location)
+                            .goto_expr;
+                            let place_goto_expr = if pt.is_generator() {
+                                place_goto_expr.member("direct_fields", &self.symbol_table)
+                            } else {
+                                place_goto_expr
+                            };
```

Comment:
This happens more than once. Utility fn?

Reply:
Factored out into a utility function.